### PR TITLE
feat(cdn): auth from cds api jwt

### PR DESCRIPTION
**Pull request by richardlt**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14477.0 ✘  
* Build
  * Build and Package All ✘  
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✘  
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14487.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14508.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by yesnault — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14510.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package and Analyze
  * Docker ✔ 
  * Sonar ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘  
  * Standalone tests 1 -   
  * Standalone tests 2 -   
  * Standalone tests 3 -

File: engine/cdn/auth.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+func (s *Service) jwtMiddleware(ctx context.Context, w http.ResponseWriter, req *http.Request, rc *service.HandlerConfig) (context.Context, error) {
+	ctx, end := telemetry.Span(ctx, "router.jwtMiddleware")
+	defer end()
+
+	v := authentication.NewVerifier(s.ParsedAPIPublicKey)
+	return service.JWTMiddleware(ctx, w, req, rc, v.VerifyJWT)
+}
+
+func (s *Service) itemAccessMiddleware(ctx context.Context, w http.ResponseWriter, req *http.Request, rc *service.HandlerConfig) (context.Context, error) {
+	vars := mux.Vars(req)
```
Comment (fsamin) [inline engine/cdn/auth.go:30]: add telemetry.Span

File: engine/cdn/auth.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+	apiRef, ok := vars["apiRef"]
+	if !ok {
+		return ctx, sdk.WithStack(sdk.ErrUnauthorized)
+	}
+
+	return s.itemAccessCheck(ctx, itemType, apiRef)
+}
+
+func (s *Service) itemAccessCheck(ctx context.Context, itemType sdk.CDNItemType, apiRef string) (context.Context, error) {
+	// Check for session based on jwt from context
```
Comment (fsamin) [inline engine/cdn/auth.go:48]: add telemetry.Span

File: engine/cdn/auth.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	if exists {
+		return ctx, nil
+	}
+
+	item, err := item.LoadByAPIRefHashAndType(ctx, s.Mapper, s.mustDBWithCtx(ctx), apiRef, itemType)
+	if err != nil {
+		return ctx, sdk.NewErrorWithStack(err, sdk.ErrNotFound)
+	}
+
+	if err := s.Client.WorkflowLogAccess(item.APIRef.ProjectKey, item.APIRef.WorkflowName, sessionID); err != nil {
```
Comment (fsamin) [inline engine/cdn/auth.go:71]: i could be nice to ba able to give this func a ctx to be able to track the telemetry.span through cdn + api

File: engine/cdn/item_logs_handler.go [hunk 4]
```diff
@@ -69,15 +67,15 @@ func (s *Service) getItemDownloadHandler() service.Handler {
...
 		vars := mux.Vars(r)
 		itemType := sdk.CDNItemType(vars["type"])
-		if err := itemType.Validate(); err != nil {
-			return err
-		}
-		token, err := s.checkAuth(r)
+		apiRef := vars["apiRef"]
+
+		// User can give a refresh delay in seconds, Refresh header value will be set if item is not complete
+		refreshDelay, err := strconv.ParseInt(r.FormValue("refresh"), 10, 64)
```
Comment (fsamin) [inline engine/cdn/item_logs_handler.go:73]: you can use `FormInt`

**Review by fsamin — CHANGES_REQUESTED** [review]

**Review by sguiheux — APPROVED** [review]

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#14533.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Docs -   
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Review by fsamin — APPROVED** [review]

**Pull request merged**